### ISSUE TITLE
movt/movf improperly restricted to mips32r2

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#215]
While looking at another issue, I found that opdebug tests of cctest/test-run-machops fail on mips32r1 boards (broadcom).

Example tests that fail are test-run-machops/RunFloat64MaxP, RunFloat64MinP, RunFloat32MaxP, RunFloat32MinP

The problem is that the the FP mov instruction DHECK mips32r2:

```
void Assembler::movt_d(FPURegister fd, FPURegister fs, uint16_t cc) {
  DCHECK(IsMipsArchVariant(kMips32r2));
  FPURegister ft;
  ft.reg_code = (cc & 0x0007) << 2 | 1;
  GenInstrRegister(COP1, D, ft, fs, fd, MOVF);
}
```

These instructions should also allow mips32r1. Note that release builds work and pass tests just fine, so the instructions work on r1, the problem is only the DCHECK.

We have another problem in that we officially support 'loongson' and these instructions were introduced in mips32, I believe.

Let's get a quick CL done to allow these new instruction on mips32r1. Please look thru all the assembler instructions for this.

These were added in 0f93a456 Reland "MIPS: Add float instructions and test coverage, part one"

I believe that the first time any of these were used was in 8ae7c9ab MIPS: [turbofan] Properly implement Float64/32 Min/Max instructions.

(And we missed these fails because we do not have a debug r1 buildbot. I'll get that taken added to the bots soon.)

**Comment by djpesic** [comment]
PTAL https://codereview.chromium.org/1448843002

**Comment by paul99** [comment]
Fixed in: https://chromium.googlesource.com/v8/v8/+/d40445884939341c33a78d64962686ddee2a733e
https://codereview.chromium.org/1448843002

In addition to the FloatMin/Max, this fixed many other TF related tests from asm.js and test suites. There were ~108 failures due to this.

after: https://v8.mips.com/bbv8/builders/Target%20-%20Debug%20-%20edge%20-%20mipsr1/builds/2
before: https://v8.mips.com/bbv8/builders/Target%20-%20Debug%20-%20edge%20-%20mipsr1/builds/1

We now fail only classes-subclass-builtins (https://github.com/paul99/v8m-rb/issues/217)
and regress-2596 (https://github.com/paul99/v8m-rb/issues/206).

Closing.
